### DISCUSSION TITLE
从net换过来的一些困难

### DISCUSSION THREAD

**Discussion by sqq135** [hooke007/mpv_PlayKit#409] [mpv 通用]
如题，被别人推荐入坑的是net_CM的DW分支版本，用的过程中有其他需要，就研究了007大佬写的文档，虽然并不是对net写的，但也能用。知道因为一些原因，net_CM版本不再更新，但暂时满足需要，就没想折腾去lazy版本。

但是最近发现用的net分支显示字幕有问题（Net_CM原版同样有问题，但Net最新版有一部分正常），再加上之前也有些没能实现的需求，就想从net迁移到lazy来，想要保留这一年多不断修改的快捷键，有几个快捷键功能实在是不会迁移，研究了一下午的uosc也没有找到，开始怀疑是不是需要再写脚本（这就超出能力范围了）。

- CTRL+a     script-binding cycle_adevice/back #menu: 音频 > 上个输出设备
- CTRL+e     cycle secondary-sub-override;show-text 双语模式:${secondary-sub-override} #menu: 字幕 > 切换 次字幕弹幕/双语模式
- E          cycle secondary-sid           #menu: 字幕 > 切换 次字幕
- e          script-message-to mpvnet cycle-subtitles #menu: 字幕 > 切换 主字幕

分别是这几个从net复制过来的功能，不知道该在lazy上如何实现。

还有，add chapter 1时会在最上方显示进度条，相比现在seek +5时显示的数字时间，感觉更直观。尝试这么设置，却只显示最底下的进度条，想知道如何修改。

- RIGHT             seek  4; script-binding uosc/flash-timeline
- SHIFT+LEFT seek -25; script-message-to uosc flash-elements progress

如果大佬们有时间，麻烦看一眼。

**Comment by hooke007** [comment]
> `script-binding cycle_adevice/back`

现在不需要这个，界面里就有按钮

> `cycle secondary-sub-override`

https://mpv.io/manual/master/#options-secondary-sub-ass-override

> `cycle secondary-sid`

这是mpv的原始命令

> `script-message-to mpvnet cycle-subtitles`

没看过原始代码是什么，但我猜测功能和它功能一致
`cycle sid`

**Reply by hooke007** [reply]
seek 不显示osd是因为我禁掉了 https://github.com/hooke007/MPV_lazy/blob/d41238313ba8c8c9f7cb77cb96cdec8977c736f9/portable_config/mpv.conf#L63

**Reply by sqq135** [reply]
感谢指导，这个脚本之前没有看到，现在好了，可以研究下了

Q1
```
e script-message-to mpvnet cycle-subtitles #menu: 字幕 > 切换 主字幕

这个是想切换字幕，是这个吗？
- <KEY>   script-binding input_plus/trackS_back         # 上一个字幕轨道...
```

Q2
```
script-binding cycle_adevice/back这个功能，你说界面里有按钮，是这个吗？
 - <KEY>   script-binding input_plus/adevice_back        # 上一个音频输出设备
```

Q3
```
我给改成了
-  osd-on-seek = msg-bar
就解决了

但有个问题就是osd-on-seek = msg-bar这其中的msg是显示毫秒的，如何能取消毫秒

是portable_config/script-opts/uosc.conf中的这个吗？
- # 显示时间码的亚秒，精确等效到秒的小数点后的位数。默认 0
time_precision=0

是需要修改这个吗？
```

Q5
```
最后想问下安装的过程，我是直接安装标准包，然后再粘贴进去hook和vslite两个包，虽然有些重复的需要替换，都一样。是这样安装的吗？

但是没有script-opts文件夹，也没有portable_config/script-opts/uosc.conf和portable_config/scripts/input_plus.lua，这些属于是进阶项目，请问有其他类似需要了解的吗？

最后问题是script都是lua是主文件，script-opts内的conf是快捷键吗？script-opts.conf这个文件net中并不存在，功能是类似script-opts内的conf吗？
```

这么多问题实在不好意思，我正好属于那种只会一点表面，本质是一点不会的，所以才这么多问题。（markdown也是最近刚学，如果排版不好那就麻烦费眼了）

**Reply by sqq135** [reply]
Q5附加
script-opts这个文件夹也都自己创建了，就是稍微有点奇怪

**Reply by zhongfly** [reply]
不是代码的话，就不要用那个代码块的语法了，代码块是不会根据网页宽度自动换行的，对于长串的文字看起来不方便。

**Reply by hooke007** [reply]
q1 切字幕就是原始命令 `cycle sid`
q2 你不至于分不清"按钮"和"快捷键"的区别吧。。。。右键菜单里找条目；当然你找到的那东西也能用
q3q5 我不用script-opts文件夹 https://github.com/hooke007/MPV_lazy/discussions/126

问题太多的话建议从头看文档吧，我也看花了

**Reply by sqq135** [reply]
学到了，是应该用引用吗，试一下效果

>最后想问下安装的过程，我是直接安装标准包，然后再粘贴进去hook和vslite两个包，虽然有些重复的需要替换，都一样。是这样安装的吗？
>
> 但是没有script-opts文件夹，也没有portable_config/script-opts/uosc.conf和portable_config/scripts/input_plus.lua，这些属于是进阶项目，请问有其他类似需要了解的吗？
>
>最后问题是script都是lua是主文件，script-opts内的conf是快捷键吗？script-opts.conf这个文件net中并不存在，功能是类似script-opts内的conf吗？

**Reply by hooke007** [reply]
> 是这样安装的吗？

是的，补丁包。以前给游戏打补丁玩过的吧，一样的流程。

核心快捷键只由 input conf 指定。script-opts内的是脚本选项，上一个回答里讲了

**Reply by hooke007** [reply]
从零建议把这个过一遍 https://hooke007.github.io/unofficial/mpv_start.html
至少对mpv的常见用法功能有个系统了解。

**Reply by sqq135** [reply]
解决所有问题了，感谢。

哈哈，这个入门篇刚开始用mpv时看过，因为是用配置好的，且当时没基础，就没怎么记住，再看一遍又了解不少，常看常新。

至于按钮，常见操作都用快捷键了，刚看到没想起来

以下是问题总结

>Q1
>e                cycle sid                                       #! 字幕 > 切换 字幕 
>可惜只有字幕标号，没有类似net的（2/4）和字幕类型但也无所谓了

>Q2
> - <KEY>   script-binding input_plus/adevice_back        # 上一个音频输出设备

>Q3
>带有毫秒不好看，就改这个
> #osd-fractions = yes

>Q5
>从零建议把这个过一遍 https://hooke007.github.io/unofficial/mpv_start.html
>至少对mpv的常见用法功能有个系统了解。
>一点不错，全解决了

**Reply by sqq135** [reply]
最后一个问题（两个问句），不知道这块我理解有没有错。

直接安装的mpv.conf中有这样的代码

 ####################
 # 脚本 滤镜 着色器 #
 ####################

.....
 glsl-shaders-append = "~~/shaders/AMD_FSR_EASU_rgb.glsl"   # 注意：此参数当前在 script-opts.conf 文件中存在关联处理（参见 “保存全局属性并恢复” 部分）


最后一句是追加滤镜，这个追加滤镜这么优秀，默认就用上了？2070也不是amd显卡啊，我看滤镜中也没有特别推荐啊？

以下是保存script-opts.conf 

####################
# 全局属性保存恢复 #
####################

#script-opts-append = save_global_props-save_mode=0
 script-opts-append = save_global_props-props=volume,glsl-shaders
#script-opts-append = save_global_props-cache_dir=~~/_cache/

就仅仅是保存同文件上次播放的 声音、着色器 状态吗？

**Reply by hooke007** [reply]
着色器不是滤镜。
作为默认和优秀没关系。
easu原理上近似lanczos

**Reply by hooke007** [reply]
保存全局属性的意思是，维持这个属性的状态。
因为mpv没有自带保存选项改动的功能

**Reply by sqq135** [reply]
不好意思，使用中还有个问题没能解决：只要鼠标放在窗口上方或者下方（比如进度条，显示缩略图）的功能区，osc就不会自动隐藏。想要只要鼠标不动，即使放在功能区，也能自动隐藏，应该调整哪个参数啊？

感觉像是这个里面，但是没有找到。portable_config/script-opts/osc.conf

**Reply by hooke007** [reply]
我不使用内置osc;另，我对这个功能没有印象

**Reply by sqq135** [reply]
哈，其实快捷键用多了也不怎么需要osc，但是还是太喜欢thumbfast这个功能了。我在uosc的讨论区提问了，希望能给个答复

我把所有参数放进gpt里，也没找到相关的
